src/thor/worker.cc

```diff
@@ -15,6 +15,10 @@
 
 #include <boost/property_tree/ptree.hpp>
 
+#ifdef ENABLE_GDAL
+#include <gdal_priv.h>
+#endif
+
```

Comment:
`tyr` has no way to maintain state, so `thor_worker_t` will hold the geotiff driver and pass an opaque pointer to it to the serializer function (note that [GDAL is not thread-safe](https://gdal.org/user/multithreading.html#gdal-api-re-entrant-but-not-thread-safe))